classes/class-wc-rest-connect-address-normalization-controller.php

```diff
@@ -19,6 +19,16 @@ public function post( $request ) {
 
 		unset( $address['company'], $address['phone'] );
 
+		/**
+		 * We only want to validate the name field if the API
+		 * version is greater than 5. If it's 5 or less
+		 * don't validate the name field.
+		 */
+		if ( WOOCOMMERCE_CONNECT_SERVER_API_VERSION <= 5 ) {
+			$name = $address['name'];
+			unset( $address['name'] );
+		}
```

Comment:
I'm fairly certain these version checks aren't necessary, as per our chat.

```suggestion
		$name = $address['name'];
		unset( $address['name'] );
```